openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -0,0 +1,138 @@
+        return
+
+    account_settings = get_account_settings(user)
+    course_key = CourseKey.from_string(unicode(course_key))
+    course = modulestore().get_course(course_key, depth=0)
+    course_display_name = course.display_name
+    branded_logo = dict(title='Logo', path=settings.NOTIFICATION_EMAIL_EDX_LOGO, cid=str(uuid.uuid4()))
+    tracking_context = tracker.get_tracker().resolve_context()
+    tracking_id = str(tracking_context.get('user_id'))
+    client_id = str(tracking_context.get('client_id'))
+    events = '&t=event&ec=email&ea=open'
+    tracking_pixel = 'https://www.google-analytics.com/collect?v=1&tid' + tracking_id + '&cid' + client_id + events
+    site_name = microsite.get_value('SITE_NAME', settings.SITE_NAME)
+    dashboard_url_path = reverse('dashboard')
+    courses_url_path = reverse('courses') + "?type=credit"
+    dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
+    dashboard_link = urlparse.urlunparse(dashboard_link_parts)
+    credit_course_link_parts = ("https", site_name, courses_url_path, '', '', '')
+    credit_course_link = urlparse.urlunparse(credit_course_link_parts)
+    context = {
+        'full_name': account_settings['name'],
+        'platform_name': settings.PLATFORM_NAME,
+        'course_name': course_display_name,
+        'branded_logo': branded_logo['cid'],
+        'dashboard_link': dashboard_link,
+        'credit_course_link': credit_course_link,
+        'tracking_pixel': tracking_pixel,
+    }
+
+    # create the root email message
+    notification_msg = MIMEMultipart('related')
+    # add 'alternative' part to root email message to encapsulate the plain and
+    # HTML versions, so message agents can decide which they want to display.
+    msg_alternative = MIMEMultipart('alternative')
+    notification_msg.attach(msg_alternative)
+    # render the credit notification templates
+    subject = None
+    if email_type == "credit_eligibility":
+        subject = _("Course Credit Eligibility")
+
+        # add alternative plain text message
+        email_body_plain = render_to_string('credit_notifications/credit_eligibility_email.txt', context)
+        msg_alternative.attach(MIMEText(email_body_plain, _subtype='plain'))
+
+        # add alternative html message
+        email_body = with_inline_css(
+            render_to_string("credit_notifications/credit_eligibility_email.html", context)
+        )
+        msg_alternative.attach(MIMEText(email_body, _subtype='html'))
+
+        # add images
+        logo_image = attach_image(branded_logo, 'Header Logo')
+        if logo_image:
+            notification_msg.attach(logo_image)
+
+    from_address = microsite.get_value('default_from_email', settings.DEFAULT_FROM_EMAIL)
+    to_address = account_settings['email']
+
+    # send the root email message
+    msg = EmailMessage(subject, None, from_address, [to_address])
+    msg.attach(notification_msg)
+    msg.send()
+
+
+def with_inline_css(html_without_css):
```

Comment:
The output of this function seems like something we should cache, so we don't need to read from disk every time we send an email.